bundles/org.openhab.binding.qolsysiq/src/main/java/org/openhab/binding/qolsysiq/internal/handler/QolsysIQPartitionHandler.java

```diff
@@ -0,0 +1,357 @@
+        updatePartitionStatus(event.armingType);
+        updateDelay(event.delay == null ? 0 : event.delay);
+        Partition partitionCache = this.partitionCache;
+        if (partitionCache != null) {
+            partitionCache.status = event.armingType;
+        }
+    }
+
+    protected void errorEvent(ErrorEvent event) {
+        cancelErrorDelayJob();
+        updateState(QolsysIQBindingConstants.CHANNEL_PARTITION_ERROR_EVENT, new StringType(event.description));
+        errorFuture = scheduler.schedule(this::clearErrorEvent, CLEAR_ERROR_MESSSAGE_TIME, TimeUnit.SECONDS);
+    }
+
+    protected void secureArmInfoEvent(SecureArmInfoEvent event) {
+        setSecureArm(event.value);
+    }
+
+    public void zoneAddEvent(ZoneAddEvent event) {
+        discoverZone(event.zone);
+    }
+
+    protected void updatePartition(Partition partition) {
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+        this.partitionCache = partition;
+        updatePartitionStatus(partition.status);
+        setSecureArm(partition.secureArm);
+        if (partition.status != PartitionStatus.ALARM) {
+            updateAlarmState(AlarmType.NONE);
+        }
+        synchronized (zones) {
+            zones.clear();
+            zones.addAll(partition.zoneList);
+            zones.forEach(z -> {
+                QolsysIQZoneHandler zoneHandler = zoneHandler(z.zoneId);
+                if (zoneHandler != null) {
+                    zoneHandler.updateZone(z);
+                }
+            });
+        }
+        discoverChildDevices();
+    }
+
+    protected @Nullable Zone getZone(Integer zoneId) {
+        synchronized (zones) {
+            return zones.stream().filter(z -> z.zoneId.equals(zoneId)).findAny().orElse(null);
+        }
+    }
+
+    private void refresh() {
+        QolsysIQPanelHandler panel = panelHandler();
+        if (panel != null) {
+            panel.refresh();
+        }
+    }
+
+    private void updatePartitionStatus(PartitionStatus status) {
+        updateState(QolsysIQBindingConstants.CHANNEL_PARTITION_ARM_STATE, new StringType(status.toString()));
+        cancelErrorDelayJob();
+        if (status == PartitionStatus.DISARM) {
+            updateAlarmState(AlarmType.NONE);
+            updateDelay(0);
+        }
+    }
+
+    private void setSecureArm(Boolean secure) {
+        Map<String, String> props = new HashMap<String, String>();
+        props.put("secureArm", String.valueOf(secure));
+        getThing().setProperties(props);
+    }
+
+    private void updateDelay(Integer delay) {
+        cancelExitDelayJob();
+        if (delay <= 0) {
+            updateState(QolsysIQBindingConstants.CHANNEL_PARTITION_COMMAND_DELAY, new DecimalType(0));
+            return;
+        }
+
+        final long endTime = System.currentTimeMillis() + (delay * 1000);
+        delayFuture = scheduler.scheduleAtFixedRate(() -> {
```

Comment:
It is not recommended to use `scheduleAtFixedRate`. Can't you use `scheduleWithFixedDelay` there ?

Reply:
So, this is designed to emit a state change exactly every 1 second (acts like a clock), so i think it's one of the few cases where `scheduleWithFixedDelay` is not substitutable.

Reply:
Ok